backend/eventloop/application_event_loop/application_event_runner_deployments_test.go

```diff
@@ -0,0 +1,97 @@
+package application_event_loop
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Application Event Runner Deployments", func() {
+	Context("createSpecField() should generate a valid argocd Application", func() {
+		It("Input spec is converted to an argocd Application", func() {
+			input := fakeInputSpec(false, false)
+			application, err := createSpecField(input)
+			Expect(err).To(BeNil())
+			Expect(application).To(Equal(validApplication()))
+		})
+
+		It("Sanitize illegal characters from input", func() {
+			input := fakeInputSpec(false, true)
+			application, err := createSpecField(input)
+			Expect(err).To(BeNil())
+			Expect(application).To(Equal(validApplication()))
+		})
+
+		It("Input spec with automated enabled should set automated sync policy", func() {
+			input := fakeInputSpec(true, false)
+			application, err := createSpecField(input)
+			Expect(err).To(BeNil())
+			Expect(application).To(Equal(validApplicationWithAutomatedSync()))
+		})
+	})
+})
+
+func fakeInputSpec(automated, unsanitized bool) argoCDSpecInput {
```

Comment:
The name of the function is to generic, as `fakeInputSpec` could mean anything within the context of `application_event_loop` package. Just a alternative suggestion:

```suggestion
func getFakeArgoCDSpecInput(automated, unsanitized bool) argoCDSpecInput {
```

Reply:
Thanks. Updated the name